cime_config/cesm/machines/config_compilers.xml

```diff
@@ -69,7 +69,6 @@ using a fortran linker.
      compilers -->
 <compiler>
   <CPPDEFS>
-    <append> -D<env>OS</env> </append>
```

Comment:
This caused problems on my mac with gnu: it leads to a bare "-D" in the compilation line, leading to a compilation error. @jedwards4b - It looks like you added this line very recently, in trying to get the unit tests to work on yellowstone, but it appears not to be needed for that purpose (scripts_regression_tests pass without it). If you feel it's needed in some cases, then we'll need to find some other tweak to allow this to work - e.g., by always ensuring that the OS environment variable is defined.